src/useForm.tsx

```diff
@@ -64,7 +64,7 @@ function getInitialState<T>(fieldDefs: FieldDefinitions<T>): FieldsState<T> {
 
 function createFields<T>(
   state: FieldsState<T>,
-  setState: (state: FieldsState<T>) => void
+  setState: (f: (state: FieldsState<T>) => FieldsState<T>) => void
```

Comment:
The `setState` function React gives us is a union type, so here I just switch the param type to the other side of the union.